src/connector/src/sink/file_sink/gcs.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::collections::{BTreeMap, HashMap};
+
+use anyhow::anyhow;
+use opendal::layers::{LoggingLayer, RetryLayer};
+use opendal::services::Gcs;
+use opendal::Operator;
+use serde::Deserialize;
+use serde_with::serde_as;
+use with_options::WithOptions;
+
+use super::opendal_sink::FileSink;
+use crate::sink::file_sink::opendal_sink::OpendalSinkBackend;
+use crate::sink::{Result, SinkError, SINK_TYPE_APPEND_ONLY, SINK_TYPE_OPTION, SINK_TYPE_UPSERT};
+
+#[derive(Deserialize, Debug, Clone, WithOptions)]
+pub struct GcsCommon {
+    #[serde(rename = "gcs.bucket_name")]
+    pub bucket_name: String,
+
+    /// The base64 encoded credential key. If not set, ADC will be used.
+    #[serde(rename = "gcs.credential")]
+    pub credential: Option<String>,
+
+    /// If credential/ADC is not set. The service account can be used to provide the credential info.
+    #[serde(rename = "gcs.service_account", default)]
+    pub service_account: Option<String>,
+
+    /// The directory where the sink file is located
+    #[serde(rename = "gcs.path", default)]
+    pub path: String,
+
+    #[serde(flatten)]
+    pub unknown_fields: HashMap<String, String>,
```

Comment:
Is `unkonwn_fields` neccessary? I only see it exist in gcs sink, not in fs and s3 sink.